### ISSUE TITLE
mzucker's Noteshrink integration

### ISSUE THREAD

**Issue by CfKu** [jbarlow83/OCRmyPDF#436]
Just a small suggestion: Have you already had a look on mzucker's algorithm to subtract background before executing unpaper (aka noteshrink)?
* https://github.com/mzucker/noteshrink / article: https://mzucker.github.io/2016/09/20/noteshrink.html
* https://github.com/gosha20777/NoteShrink

**Comment by jbarlow83** [comment]
That looks really interesting.

This is the sort of thing I'd like to use the forthcoming plugin architecture for, if/when I get a chance to finish it.

**Comment by CfKu** [comment]
Hi @jbarlow83,
In the manual I read about the plugins (https://ocrmypdf.readthedocs.io/en/latest/plugins.html). Do you think it is feasible to call the noteshrink scripts?

**Comment by jbarlow83** [comment]
Yes, you could write a plugin that hooks `filter_page_image` for example.
